File: packages/amplify_auth/android/src/main/kotlin/com/amazonaws/amplify/types/FlutterSignUpRequest.kt
package com.amazonaws.amplify.amplify_auth

data class FlutterSignUpRequest(val map: HashMap<String, *>) {
    val username: String? = map["username"] as String?;
    val password: String = map["password"] as String;
    val userAttributes: HashMap<String, String> = map["userAttributes"] as HashMap<String, String>;
    val providerOptions: HashMap<String, *>? = map["providerOptions"] as HashMap<String, *>?;
}

File: packages/amplify_auth/ios/Classes/FlutterSignUpRequest.swift
//
//  FlutterSignUpRequest.swift
//  amplify_auth
//
//  Created by Noyes, Dustin on 7/2/20.
//

import Foundation
struct FlutterSignUpRequest {
  var username: String?;
  var password: String?;
  var usernameAttribute: String?;
  var userAttributes: Dictionary<String, String> = [:];
  var validationData: Dictionary<String, String>?;
  
  init(dict: NSMutableDictionary){
    self.username = dict["username"] as! String?
    self.password = dict["password"] as! String?
    self.usernameAttribute = dict["usernameAttribute"] as! String?
    self.userAttributes = dict["userAttributes"] as! Dictionary<String, String>
    self.validationData = dict["validationData"] as! Dictionary<String, String>?
  }
}
